convert.lua

```diff
@@ -168,8 +176,8 @@ function convert_prot.dissector(buffer, pinfo, tree)
         return
     end
 
-    -- Assume Convert Header sits in the SYN. Otherwise Ignore.
-    if is_convert_syn() then
+    -- Assume Convert Header sits in the SYN or the ACK. Otherwise Ignore.
+    if (is_convert_syn() or is_convert_ack()) then
```

Comment:
but then here, this condition will only ignore the SYN+ACK.
Maybe we should mark the packet as "convert" only for the first packet carrying data in this connection, no?

So I would say:
* if there is no data, return → `msg_length == 0`
* if it is not the first packet in this direction carrying data if, return → `is_past_convert_msg()`
* we don't need any other checks
  * if you are here, it is either a Convert packet or we cannot parse it is another protocol
  * so we should always update `convert_end_pkt_num[stream_dir_key]` and then try parsing.

WDYT? :)

Reply:
+1 for @matttbe 's proposal. That's easy enough and should do what we want (of course it won't handle the case where the Convert message is split over multiple segments, but that should be a rare corner case).

I guess we no longer need the mark_stream_as_convert() function and the related hash-map. I think this was only used to recognize the need to parse the segment in the return direction (Converter->Client).

Reply:
(later) We can also parse up to X bytes or up to have the end of convert message → we keep the number of bytes we parsed so far and a flag to say if we need to parse more.

Reply:
It still does not cover the case where the Convert message is retransmitted, but I guess that doesn't matter.

Reply:
Good point!

We could probably add support for this (and split messages) later.